eclass/kernel-install.eclass

```diff
@@ -234,25 +279,44 @@ kernel-install_test() {
 		--no-hostonly \
 		--kmoddir "${modules}" \
 		"${T}/initrd" "${version}" || die
-	# get a read-write copy of the disk image
-	cp "${DISTDIR}/tinycorelinux-${TCL_VER}-${ARCH}.qcow2" \
-		"${T}/fs.qcow2" || die
+
+	kernel-install_create_qemu_image "${T}/fs.img"
 
 	cd "${T}" || die
+
 	local qemu_extra_args=
-	[[ ${qemu_arch} == x86_64 ]] && qemu_extra_args='-cpu max'
+	local qemu_extra_append=
+	local qemu_mem="512M"
+
+	case ${qemu_arch} in
+		aarch64)
+			qemu_extra_args="-M virt -cpu cortex-a57 -smp 1"
+			qemu_extra_append="console=ttyAMA0"
+			;;
+		i386|x86_64)
+			qemu_extra_args="-cpu max"
+			qemu_extra_append="console=ttyS0,115200n8"
+			;;
+		ppc64)
+			qemu_extra_args="-nodefaults"
+			;;
+		*)
+			:
+			;;
+	esac
+
 	cat > run.sh <<-EOF || die
 		#!/bin/sh
 		exec qemu-system-${qemu_arch} \
 			${qemu_extra_args} \
-			-m 256M \
-			-display none \
+			-m ${qemu_mem} \
```

Comment:
```suggestion
			-m 512M \
```

No point in the variable when it's no longer dynamic.

Reply:
it may be needed later, some arm qemu machines, like `raspi2`, require exact value to be passed.
but ok to go for now, can always be added later.